src/nsolid.h

```diff
@@ -961,7 +962,56 @@ class NODE_EXTERN Snapshot {
                           Cb&& cb,
                           Data&&... data);
 
+  /**
+   * @brief activates the heap profiler tracking objects from a specific JS
+   * thread.
+   *
+   * @param envinst SharedEnvInst of thread to take the snapshot from.
+   * @param redacted to get redacted strings in the heap snapshot.
+   * @param data variable number of arguments to be propagated to the callback.
+   * @return NSOLID_E_SUCCESS in case of success or a different NSOLID_E_
+   * error value otherwise.
+   */
+  template <typename Cb, typename... Data>
+  static int StartTrackingHeapObjects(SharedEnvInst envinst,
+                                      bool redacted,
+                                      bool trackAllocations,
+                                      uint64_t duration,
+                                      Cb&& cb,
+                                      Data&&... data);
+
+  /**
+   * @brief same as TakeSnapshot but stops tracking heap objects in the heap
+   * profiler.
+   *
+   * @param envinst SharedEnvInst of thread to take the snapshot from.
+   * @param redacted to get redacted strings in the heap snapshot.
+   * @param cb callback function with the following signature:
+   * `cb(int status, std::string snapshot, ...Data)`. It will be called from the
+   * NSolid thread.
+   * @param data variable number of arguments to be propagated to the callback.
+   * @return NSOLID_E_SUCCESS in case of success or a different NSOLID_E_
+   * error value otherwise.
+   */
+  template <typename Cb, typename... Data>
+  static int StopTrackingHeapObjects(SharedEnvInst envinst,
+                                     Cb&& cb,
+                                     Data&&... data);
```

Comment:
As we discussed, no reason to have a cb for this as the completion will be notified using the `StartTrackingHeapObjects` cb.
```suggestion
  template <typename Cb, typename... Data>
  static int StopTrackingHeapObjects(SharedEnvInst envinst);
```

Reply:
Maybe,,,
```suggestion
  static int StopTrackingHeapObjects(SharedEnvInst envinst); // No template required here
```